compiler-rt/lib/sanitizer_common/sanitizer_thread_registry.cpp

```diff
@@ -274,6 +274,11 @@ void ThreadRegistry::JoinThread(u32 tid, void *arg) {
     {
       ThreadRegistryLock l(this);
       ThreadContextBase *tctx = threads_[tid];
+
+      if (!tctx) {
+        Report("%s: Tried to join thread %u, but it is not live.", SanitizerToolName, tid);
+      }
+
       CHECK_NE(tctx, 0);
```

Comment:
can you move CHECK inside of if(!tctx), just for readablitity
to highlight that they are related